### ISSUE TITLE
its not work on bunjs

### ISSUE THREAD

**Issue by ddosakura** [yoshoku/hnswlib-node#187]
its not work on bunjs. Is it possible to be bun compatible?

also see:
https://github.com/langchain-ai/langchainjs/issues/3782
https://github.com/nmslib/hnswlib/issues/533

**Comment by yoshoku** [comment]
@ddosakura 
hnsw-node also work on Bun:

```typescript
// index.ts

import { BruteforceSearch, HierarchicalNSW } from 'hnswlib-node'

const numDimensions = 8;
const maxElements = 10;

const index = new HierarchicalNSW('l2', numDimensions);
index.initIndex(maxElements);

for (let i = 0; i < maxElements; i++) {
  const point = new Array(numDimensions);
  for (let j = 0; j < numDimensions; j++) point[j] = Math.random();
  index.addPoint(point, i);
}

const query = new Array(numDimensions);
for (let j = 0; j < numDimensions; j++) query[j] = Math.random();

const numNeighbors = 3;
const result = index.searchKnn(query, numNeighbors);

console.table(result);
```

<img width="693" alt="bun_hnswlib-node" src="https://github.com/yoshoku/hnswlib-node/assets/5562409/1bafee6a-48a6-4c8d-b53c-9994b914576a">

Please read the error message carefully in the screenshot of the issue:

```
error: Could not import hnsw-node. Please install hnsw-node as a dependency with, e. g. `npm install -S hnswlib-node`
```

The error probably occurred because hnswlib-node was not installed.

**Comment by yoshoku** [comment]
Since the original issues are closed, this issue is also closed. Thank you for your understanding.

**Comment by ddosakura** [comment]
Reasons for not work:

1. Early versions of bunjs did not auto run node-gyp scripts

also see:

https://github.com/microsoft/node-pty/issues/632#issuecomment-1982243227

This has been fixed in `bun:1.1.0+`:

https://github.com/oven-sh/bun/pull/9289

2. the official bunjs image not work

https://github.com/oven-sh/bun/issues/9807

The issue has been closed, but the latest image still reports an error

```bash
docker run --interactive --tty --rm -v $PWD:$PWD --workdir $PWD oven/bun:1.1.3-debian /bin/bash

apt update
apt install python3

rm -rf node_modules/
bun i hnswlib-node
# gyp: name 'llvm_version' is not defined while evaluating condition 'llvm_version=="0.0"' in binding.gyp while trying to load binding.gyp
```

use nodejs image:

https://github.com/oven-sh/bun/issues/9807#issuecomment-2041573065

```bash
docker run --interactive --tty --rm -v $PWD:$PWD --workdir $PWD node:20.12.1-slim /bin/bash

npm i -g bun

apt update
apt install python3 make g++

rm -rf node_modules/
bun i hnswlib-node
```

it works!
